tests/test_data_file.py

```diff
@@ -249,3 +259,322 @@ def test_data_file_topology(snap_8, tmp_path, shuffle_ids):
+    if shuffle_ids:
+        snap.id = [3, 1, 2]
+
+    # write the data file with nondefault values using lammpsio
+    snap.position = [[0.1, 0.2, 0.3], [-0.4, -0.5, -0.6], [0.7, 0.8, 0.9]]
+    snap.image = [[1, 2, 3], [-4, -5, -6], [7, 8, 9]]
+    snap.velocity = [[-3, -2, -1], [6, 5, 4], [9, 8, 7]]
+    snap.typeid = [2, 1, 2]
+    snap.mass = [3, 2, 3]
+    if atom_style in ("molecular", "full"):
+        snap.molecule = [2, 0, 1]
+    if atom_style in ("charge", "full"):
+        snap.charge = [-1, 0, 1]
+
+    _tmp = tempfile.TemporaryDirectory()
+    directory = _tmp.name
+    filename = directory + "/atoms.data"
+
+    # create the data file using lammpsio
+    lammpsio.DataFile.create(filename, snap, atom_style if set_style else None)
+    assert os.path.isfile(filename)
+
+    lmps = lammps.lammps(cmdargs=["-log", f"{directory}/log.lammps"])
+    cmds = [f"atom_style {atom_style}"]
+    cmds += [f"read_data {filename}"]
+    cmds += [f"write_data {filename}"]
+    lmps.commands_list(cmds)
+    lmps.close()
+
+    # read it back in and check
+    snap_2 = lammpsio.DataFile(filename).read()
+    assert snap_2.N == snap.N
+    assert snap_2.step is None
+    assert numpy.allclose(snap_2.box.low, snap.box.low)
+    assert numpy.allclose(snap_2.box.high, snap.box.high)
+    if snap.box.tilt is not None:
+        assert numpy.allclose(snap_2.box.tilt, snap.box.tilt)
+    else:
+        assert snap_2.box.tilt is None
+    if shuffle_ids:
+        assert snap_2.has_id()
+        assert numpy.allclose(snap_2.id, snap.id)
+    else:
+        assert not snap_2.has_id()
+    assert snap_2.has_position()
+    assert numpy.allclose(snap_2.position, snap.position)
+    assert snap_2.has_velocity()
+    assert numpy.allclose(snap_2.velocity, snap.velocity)
+    assert snap_2.has_image()
+    assert numpy.allclose(snap_2.image, snap.image)
+    assert snap_2.has_typeid()
+    assert numpy.allclose(snap_2.typeid, snap.typeid)
+    assert snap_2.has_mass()
+    assert numpy.allclose(snap_2.mass, snap.mass)
+    if atom_style in ("molecular", "full"):
+        assert snap_2.has_molecule()
+        assert numpy.allclose(snap_2.molecule, snap.molecule)
+    else:
+        assert not snap_2.has_molecule()
+    if atom_style in ("charge", "full"):
+        assert snap_2.has_charge()
+        assert numpy.allclose(snap_2.charge, snap.charge)
+    else:
+        assert not snap_2.has_charge()
+    _tmp.cleanup()
+
+
+@pytest.mark.skipif(not has_lammps, reason="lammps not installed")
+@pytest.mark.parametrize("shuffle_ids", [False, True])
+def test_data_file_topology_lammps(snap_8, tmp_path, shuffle_ids):
+    # set ids to be assigned
+    if shuffle_ids:
+        particle_id = [1, 5, 2, 6, 3, 7, 4, 8]
+        bond_id = [1, 4, 2, 5, 3, 6]
+        angle_id = [1, 3, 2, 4]
+        # Dihedral ids are automatically sorted by LAMMPS
+        dihedral_id = [1, 2]
+        # LAMMPS sorts
+        improper_id = [1, 2]
```

Comment:
Please expand this comment, it is confusing why you are shuffling `bond_id` and `angle_id`, but not `dihedral_id` and `improper_id`